test/hotspot/jtreg/serviceability/jvmti/SuspendWithObjectMonitorWait/SuspendWithObjectMonitorWait.java

```diff
@@ -97,14 +199,14 @@ public static void main(String[] args) throws Exception {
         }
 
         int timeMax = 0;
-        if (args.length == 0) {
+        if (args.length == 1) {
             timeMax = DEF_TIME_MAX;
         } else {
-            int argIndex = 0;
-            int argsLeft = args.length;
-            if (args[0].equals("-p")) {
+            int argIndex = 1;
+            int argsLeft = args.length-argIndex;
+            if (args[argIndex].equals("-p")) {
                 printDebug = true;
-                argIndex = 1;
+                argIndex = 2;
                 argsLeft--;
             }
             if (argsLeft == 0) {
```

Comment:
Because of the addition of a mandatory argument (test case), argument parsing will
be more complicated. I need to mull on this a bit and I'll post another suggestion
after this round.

Reply:
I emailed you an updated version of the current test (v07) with a better
parser. Feel free to merge it in or not.